src/components/configuration/Rustup.ts

```diff
@@ -110,7 +110,14 @@ export class Rustup {
      * Returns either the nightly toolchain chosen by the user or undefined
      */
     public getUserNightlyToolchain(): Toolchain | undefined {
-        return this._userNightlyToolchain;
+        if (this._userNightlyToolchain) {
+            return this._userNightlyToolchain;
+        }
+        let def = this.getDefaultToolchain();
```

Comment:
Could you make it `const` instead of `def` please?
Could you rename it to `defaultToolchain` as `def`, at least for me, stands for "definition"?